mail_tracking/static/src/js/failed_message/discuss.js

```diff
@@ -0,0 +1,398 @@
+/* Copyright 2019 Alexandre Díaz
+   License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). */
+odoo.define('mail_tracking.FailedMessageDiscuss', function (require) {
+    "use strict";
+
+    // To be considered:
+    //   - One message can be displayed in many threads
+    //   - A thread can be a mailbox, channel, ...
+    //   - A mailbox it's a type of thread that it's displayed on top of
+    //    the discuss menu, has a counter, etc...
+
+    var MailManagerNotif = require('mail.Manager.Notification');
+    var AbstractMessage = require('mail.model.AbstractMessage');
+    var Message = require('mail.model.Message');
+    var Discuss = require('mail.Discuss');
+    var MailManager = require('mail.Manager');
+    var Mailbox = require('mail.model.Mailbox');
+    var core = require('web.core');
+    var session = require('web.session');
+
+    var QWeb = core.qweb;
+    var _t = core._t;
+
+    /* The states to consider a message as failed message */
+    var FAILED_STATES = [
+        'error', 'rejected', 'spam', 'bounced', 'soft-bounced',
+    ];
+
+
+    AbstractMessage.include({
+
+        /**
+         *  Abstract declaration to know if a message its included on the
+         * failed mailbox. By default should be false.
+         *
+         * @returns {Boolean}
+         */
+        isFailed: function () {
+            return false;
+        },
+    });
+
+    Message.include({
+
+        /**
+         * Overrides to store information from server
+         *
+         * @Override
+         */
+        init: function (parent, data) {
+            this._isFailedMessage = data.is_failed_message;
+            this._super.apply(this, arguments);
```

Comment:
Probably `this._super()` is returning nothing, but just in case, when this one is the latest sentence on a method, it's a good rule of thumb to `return` it.

```suggestion
            return this._super.apply(this, arguments);
```

Reply:
A function with empty return and no-return will return the same canonical 'undefined' value. This is only for readability no?

Reply:
> This is only for readability no?

Yes, it's more obvious so you don't have to wonder if super() was actually returning or not.